libtenzir/builtins/commands/exec.cpp

```diff
@@ -112,7 +112,15 @@ auto exec_command(const invocation& inv, caf::actor_system& sys) -> bool {
       printer->emit(diagnostic::error(result.error()).done());
       return false;
     }
-    content = std::move(*result);
+    // Strip potential YAML front matter.
+    if (result->starts_with("---\n")) {
+      auto skip = result->find("---\n", 4);
+      if (skip == std::string::npos) {
+      }
+      content = result->substr(skip + 4);
+    } else {
+      content = std::move(*result);
+    }
```

Comment:
This will break diagnostic locations. One way to do this is to instead treat this in the tokenizer as whitespace, very similar to what we are doing for `#!`.